plonky2/src/plonk/circuit_builder.rs

```diff
@@ -219,10 +219,17 @@ impl<F: RichField + Extendable<D>, const D: usize> CircuitBuilder<F, D> {
         );
         constants.resize(gate_type.num_constants(), F::ZERO);
 
-        let index = self.gate_instances.len();
+        let gate_index = self.gate_instances.len();
 
         self.constant_generators
-            .extend(gate_type.extra_constants(index));
+            .extend(gate_type.extra_constant_wires().enumerate().map(
+                |(constant_index, input_index)| ConstantGenerator {
```

Comment:
I think this wouldn't work for a gate that uses (non-extra) constants. For example, if we wanted to add extra-constants to `ArithmeticGate`, the first two constants would be the `c0,c1` used in the arithmetic operation. So the `enumerate` here would yield incorrect indices.

I'm not sure we'll ever need extra-constants on gates already using constants, and this could be fixed by using the convention that extra-constants should be placed before other constants. But I think having `extra_constant_wires()` return a `Vec<(usize, usize)>` would be less confusing.

Reply:
Good point! Changed to `Vec<(usize, usize)>`.